### PR TITLE
Stop aeg from abusing agg ##refactor

### PR DESCRIPTION
**Pull request by condret**
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/core/agraph.c [hunk 1]
```diff
@@ -5032,14 +5032,8 @@ R_API int r_core_visual_graph(RCore *core, RAGraph *g, RAnalFunction *_fcn, int
...
 
-/**
- * @brief Create RAGraph from generic RGraph with RGraphNodeInfo as node data
- *
- * @param graph <RGraphNodeInfo>
- * @return RAGraph* NULL if failure
- */
-R_API RAGraph *create_agraph_from_graph(const RGraph/*<RGraphNodeInfo>*/ *graph) {
-	r_return_val_if_fail (graph, NULL);
+R_API RAGraph *r_agraph_create_from_graph(const RGraph *graph, RAGraphTransitionCBs *cbs) {
```
Comment (trufae) [inline libr/core/agraph.c:5035]:
```suggestion
R_API RAGraph *r_agraph_new_from_graph(const RGraph *graph, RAGraphTransitionCBs *cbs) {
```

File: test/unit/test_agraph.c [hunk 2]
```diff
@@ -19,7 +29,11 @@ bool test_graph_to_agraph() {
 	mu_assert_notnull (graph, "Couldn't create the graph");
 	mu_assert_eq (graph->nodes->length, 4, "Wrong node count");
 
-	RAGraph *agraph = create_agraph_from_graph (graph);
+	RAGraphTransitionCBs cbs = {
+		.get_title = _graph_node_info_get_title,
+		.get_body = _graph_node_info_get_body
+	};
+	RAGraph *agraph = r_agraph_create_from_graph (graph, &cbs);
```
Comment (trufae) [inline test/unit/test_agraph.c:36]:
```suggestion
	RAGraph *agraph = r_agraph_new_from_graph (graph, &cbs);
```

File: test/unit/test_agraph.c [hunk 3]
```diff
@@ -4,6 +4,16 @@
 #include <r_util.h>
 #include "minunit.h"
 
+static char *_graph_node_info_get_title (void *data) {
+	RGraphNodeInfo *info = (RGraphNodeInfo *)data;
+	return strdup (info->title);
```
Comment (trufae) [inline test/unit/test_agraph.c:9]: is strdup necessary?

File: test/unit/test_agraph.c [hunk 4]
```diff
@@ -4,6 +4,16 @@
...
+}
+
+static char *_graph_node_info_get_body (void *data) {
```
Comment (trufae) [inline test/unit/test_agraph.c:12]:
```suggestion
static char *_graph_node_info_get_body(void *data) {
```

File: test/unit/test_agraph.c [hunk 5]
```diff
@@ -4,6 +4,16 @@
 #include <r_util.h>
 #include "minunit.h"
 
+static char *_graph_node_info_get_title (void *data) {
```
Comment (trufae) [inline test/unit/test_agraph.c:7]:
```suggestion
static char *_graph_node_info_get_title(void *data) {
```

**Review by trufae — CHANGES_REQUESTED** [review]

File: test/unit/test_agraph.c (hunk 3 above)
Comment (condret) [reply test/unit/test_agraph.c:9]: yes

**Pull request merged**